ov/id_pools.go

```diff
@@ -0,0 +1,338 @@
+package ov
+
+import (
+	"encoding/json"
+
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type IdPool struct {
+	AllocatedCount int           `json:"allocatedCount,omitempty"`
+	Category       string        `json:"category,omitempty"`
+	Created        string        `json:"created,omitempty"`
+	ETAG           utils.Nstring `json:"etag,omitempty"`
+	Enabled        *bool         `json:"enabled,omitempty"`
+	FreeCount      int           `json:"freeCount,omitempty"`
+	Modified       string        `json:"modified,omitempty"`
+	Name           string        `json:"name,omitempty"`
+	PoolType       string        `json:"poolType,omitempty"`
+	Prefix         string        `json:"prefix,omitempty"`
+	RangeUris      []string      `json:"rangeUris,omitempty"`
+	TotalCount     int           `json:"totalCount,omitempty"`
+	Type           string        `json:"type,omitempty"`
+	URI            string        `json:"uri,omitempty"`
+}
+
+// Checks the range availability in the ID pool.
+func (c *OVClient) GetRangeAvailibility(poolType string, ids []string) (UpdateAllocatorList, error) {
+	var (
+		uri    = "/rest/id-pools/" + poolType + "/checkrangeavailability"
+		idList UpdateAllocatorList
+	)
+
+	q := make(map[string]interface{})
+	if len(ids) > 0 {
+		q["idList"] = ids
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return idList, err
+	}
+
+	log.Debugf("Get Id Lists %s", data)
+	if err := json.Unmarshal(data, &idList); err != nil {
+		return idList, err
+	}
+	return idList, nil
+}
+
+// Validates a set of IDs to reserve in the pool.
+func (c *OVClient) GetValidateIds(poolType string, ids []string) (UpdateAllocatorList, error) {
+	var (
+		uri    = "/rest/id-pools/" + poolType + "/validate"
+		idList UpdateAllocatorList
+	)
+
+	q := make(map[string]interface{})
+	if len(ids) > 0 {
+		q["idList"] = ids
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
+	}
+
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return idList, err
+	}
+
+	log.Debugf("Get Id Lists %s", data)
+	if err := json.Unmarshal(data, &idList); err != nil {
+		return idList, err
+	}
+	return idList, nil
+}
+
+// Gets a pool along with the list of ranges present in it
+func (c *OVClient) GetPoolType(poolType string) (IdPool, error) {
+	var (
+		uri    = "/rest/id-pools/"
+		idPool IdPool
+	)
+
+	if poolType == "" {
```

Comment:
why don't you add this check for previous functions also

Reply:
done